mbed/mbed.py

```diff
@@ -1448,8 +1448,8 @@ def check_requirements(self, show_warning=False):
         missing = []
         try:
             with open(os.path.join(req_path, req_file), 'r') as f:
-                import pip
-                installed_packages = [re.sub(r'-', '_', package.project_name.lower()) for package in pip.get_installed_distributions(local_only=True)]
+                pkg_list = pquery([python_cmd, '-m', 'pip', 'list', '-l']) or ""
```

Comment:
please note that  `-l ` is a deprecated option
it produces following warning:
>DEPRECATION: The default format will switch to columns in the future. You can use --format=(legacy|columns) (or define a format=(legacy|columns) in your pip.conf under the [list] section) to disable this warning.

consider using `--format=legacy` or `--format=column` instead

Reply:
@alzix `pip list -l` is for listing the locally installed python modules. `-l, --local                 If in a virtualenv that has global access, do not list globally-installed packages.
`

Reply:
I've also tried forcing the columns format using the `--format=columns` option and the code above is handling it just fine:

```
[snip]
                pkg_list = pquery([python_cmd, '-m', 'pip', 'list', '-l', '--format=columns']) or ""
[snip]
```

```
$ pip uninstall pyyaml
Uninstalling PyYAML-3.12:
  Would remove:
    c:\progra~1\python\lib\site-packages\pyyaml-3.12-py2.7.egg-info
    c:\progra~1\python\lib\site-packages\yaml\*
Proceed (y/n)?
Your response ('') was not one of the expected responses: y, n
Proceed (y/n)? y
  Successfully uninstalled PyYAML-3.12

$ mbed update --clean
[mbed] Synchronizing dependency references...
[mbed] Updating program "mbed-os-example-wifi" to latest revision in the current branch
[mbed] Updating library "esp8266-driver" to rev #264468722a97
[mbed] Updating library "mbed-os" to rev #c05d72c3c005 (tag: mbed-os-5.8.3)
[mbed] Updating library "wifi-x-nucleo-idw01m1" to rev #5871f7011d7f
[mbed] Auto-installing missing Python modules...

$ pip list | grep -i pyyaml
PyYAML                        3.12
```

Reply:
I think you're suggesting that we don't need to worry about the deprecation warning. That's how I took it at least.

Reply:
Yes, exactly. Sorry that I didn't explicitly said it. Was writing this in between meetings 😀